src/wasm/kotlin/README.md

```diff
@@ -1,12 +1,56 @@
-# Kotlin --> WASM particles
-
-## Build & Test
-0. Make sure the project is [setup](../../../README.md#install)!
-1. `bazel build particles/Native/Wasm:all`
-2. `npm start`
-3. Visit [localhost:8786/shells/dev-shell/?m=https://$particles/Native/Wasm/wasm.arcs](http://localhost:8786/shells/dev-shell/?m=https://$particles/Native/Wasm/wasm.arcs)
-
-## Converting Arcs Schemas to Kotlin
-Subject to change
-1. Run `$path-to-sigh/sigh schema2wasm -k $path-from-arcs-folder-to-manifest-file -f $path-from-arcs-folder-to-output-file`
-1. The Kotlin file should be in the output file you specified above.
+# Kotlin --> Wasm particles
+
+## Setup 
+
+Our code is built by Bazel. It uses Kotlin Mutliplatform to target JVM and Wasm 
+runtimes. Bazel must be installed to build and run the code and tests, which
+reside in `src/wasm/`.
+
+1. **Installing Bazel:** run the `tools/setup` script to install Bazel on Linux
+   and MacOS (or otherwise look at that script to see which version of Bazel to 
+   download and how to set it up).
+
+## Build
+
+See [here](../../../particles/Native/Wasm) or [here](../../../particles/Tutorial/Kotlin) for working examples.
+
+- Add a `BUILD` file in the relevant directory
+- Generate Kotlin Data Classes from your particle spec using the 
+  `arcs_kt_schema` build rule:
+  ```
+  arcs_kt_schema(
+      # Name of the BUILD rule (tell Bazel to build this schema using this name).
+      name = "example_schema",
+      # Input source.
+      src = "example.arcs",
+      # Optionally, specify package where entities will reside.
+      package: "arcs.example",
+  )
+  ```
+- Write your Kotlin particle(s): See [the Kotlin tutorial](../../../particles/Tutorial/Kotlin) for greater detail.
+- Add a `wasm_cc_binary` build rule for your C++ particle(s) to your `BUILD`
+  file:
+  ```
+  arcs_kt_binary(
+      # Name of the BUILD rule (tell Bazel to build this particle using this name).
+      name = "example_particle",
+      # Input Kotlin particle source files to compile.
+      srcs = ["Example.kt"],
```

Comment:
Currently, arcs_kt_binary doesn't use kt_jvm_and_wasm_library, so you'll end up with only a WASM target.

Either fix that, or use an intermediate arcs_kt_library targeted by arcs_kt_binary

Reply:
@SHeimlich Should we update the Tutorials / Demo to follow this pattern? That way, they can target the JVM in the future.

Reply:
I spawn off a new PR for this after we land these changes.